packages/hooks/src/createHooks.js

```diff
@@ -20,15 +20,17 @@ function createHooks() {
 	actions.__current = [];
 	filters.__current = [];
 
-	return {
-		addAction: createAddHook( actions ),
-		addFilter: createAddHook( filters ),
-		removeAction: createRemoveHook( actions ),
-		removeFilter: createRemoveHook( filters ),
+	const hooks = Object.create( null );
+
+	Object.assign( hooks, {
+		addAction: createAddHook( actions ).bind( hooks ),
```

Comment:
The `actions` and `filters` objects should be properties of the parent `hook` object, rather than being related together just by closure variable references.

This also avoids having to call `bind()` on the return value of every `create*` function. The `bind` call will still be there, but hidden inside the `create*` implementation.

```js
addAction: createAddHook( hooks, 'actions' );
```
etc.

Reply:
Do you think it'd be better update it here, or separately in a follow-up PR?

Reply:
@yuliyan I think making this update will also help fix the type checking errors reported by @adamsilverstein (`'this' implicitly has type 'any'`). So you probably want to do it right away 🙂